components/modules/qrcodegen.c

```diff
@@ -0,0 +1,67 @@
+#include "module.h"
+#include "lauxlib.h"
+#include "lmem.h"
+
+#include "qrcodegen.h"
+
+// qrcodegen.encodeText(text, minVer, maxVer, ecl, mask, boostEcl)
+// all params except text are optional
+static int encodeText(lua_State *L)
+{
+  const char *text = luaL_checkstring(L, 1);
+
+  int min_ver = luaL_optint(L, 2, 1);
+  luaL_argcheck(L, min_ver >= qrcodegen_VERSION_MIN && min_ver <= qrcodegen_VERSION_MAX,
+                2, "minVer must be 1-40");
+
+  int max_ver = luaL_optint(L, 3, 40);
+  luaL_argcheck(L, max_ver >= min_ver && max_ver <= qrcodegen_VERSION_MAX,
+                3, "maxVer must be 1-40");
+
+  int ecl = luaL_optint(L, 4, qrcodegen_Ecc_LOW);
+  int mask = luaL_optint(L, 5, qrcodegen_Mask_AUTO);
+  bool boost = lua_toboolean(L, 6);
+
+  const size_t buf_len = qrcodegen_BUFFER_LEN_FOR_VERSION(max_ver);
+  uint8_t *result = (uint8_t *)luaM_malloc(L, buf_len * 2); 
+  uint8_t *tempbuf = result + buf_len;
+
+  bool ok = qrcodegen_encodeText(text, tempbuf, result, ecl, min_ver, max_ver, mask, boost);
+  if (ok) {
+    lua_pushlstring(L, (const char *)result, buf_len);
```

Comment:
This assumes that qrcodegen's result doesn't exceed the maximum string length limit of the Lua VM (4095). Is this correct?

Reply:
The max size of a QRCode is `qrcodegen_BUFFER_LEN_MAX` which is 3918 so we just squeeze in under that limit, although personally I always unset that limit in my init.lua script since I find it's not helpful on the ESP32.